ArcGISRuntimeSDKQt_SampleViewers/ArcGISRuntimeSDKQt_Samples/GAnalytics.cpp

```diff
@@ -0,0 +1,164 @@
+// [Legal]
+// Copyright 2022 Esri.
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// [Legal]
+
+#include "GAnalytics.h"
+
+#include <QDebug>
+#include <QJsonDocument>
+#include <QJsonObject>
+#include <QNetworkAccessManager>
+#include <QNetworkReply>
+#include <QOperatingSystemVersion>
+#include <QQmlEngine>
+#include <QSettings>
+#include <QUuid>
+
+#if defined (GANALYTICS_API_KEY) && defined (GANALYTICS_STREAM_ID)
+GAnalytics::GAnalytics(QObject* parent /* = nullptr */):
+  QObject(parent),
+  m_apiSecret(QUOTE(GANALYTICS_API_KEY)),
+  m_measurementId(QUOTE(GANALYTICS_STREAM_ID))
+{
+}
+#endif // GANALYTICS_API_KEY && GANALYTICS_STREAM_ID
+
+GAnalytics::GAnalytics(QObject* parent /* = nullptr */):
+  QObject(parent),
+  m_apiSecret(""),
+  m_measurementId("")
+{
+}
+
+GAnalytics::~GAnalytics()
+{
+  endSession();
+}
+
+void GAnalytics::init()
+{
+  if (m_apiSecret == "" || m_measurementId == "")
```

Comment:
Use canonical QString isEmpty.
```suggestion
  if (m_apiSecret.isEmpty() || m_measurementId.isEmpty())
```